### DISCUSSION TITLE
"Command log" prompt for passphrase doesn't work

### DISCUSSION THREAD

**Discussion by ljfp** [jesseduffield/lazygit#4056] [Q&A] [answered]
When I open a local repository with Lazygit, and that local repository is synched with a remote, I get the following error on the Command log tab:
![screenshot-2024-11-12-07-35-47](https://github.com/user-attachments/assets/92fcdda4-8667-4af3-a303-7c72e79ce0c3)

It seems that it tries to fetch from origin, but I never get the prompt to write my SSH key passphrase.
This problem doesn't happen when I push or pull. In those cases I got a pop-up with an interactive screen to insert my passphrase. But on the Command log, for some reason, I can't get the same pop-up to show up.

**Comment by ljfp** [comment]
This is the pop-up I'm expecting to see:
![screenshot-2024-11-12-07-41-24](https://github.com/user-attachments/assets/409f5e93-fa91-48bd-9dca-a6fada4b39a0)

Instead, I got the error message displayed on the post. I only get that pop-up when pushing or pulling from remote. Is there any way to fix the Command log to trigger the pop-up instead of having a fatal failure?

**Reply by stefanhaller** [reply, marked as answer]
Background fetching is done regularly every minute. Do you really want to enter your pass phrase every minute? (That was a rhetorical question. No you don't. 😄)

The solution here is to install a pass phrase agent such as ssh-agent so that you have to type the phrase only once per login. Background fetching then just works, and you won't have to type it when pulling/pushing either. Since you didn't say what platform you're on, it's hard to give more concrete guidance, but the information is fairly easy to find on the Internet.

**Reply by ljfp** [reply]
Thanks Stefan! That makes sense, displaying the pop-up for _every_ background fetch operation would be annoying. I added my passphrase to my ssh-agent and now the Command log is working like a charm :)
